### ISSUE TITLE
Revamp maintenance mode

### ISSUE THREAD

**Issue by rebeccacremona** [harvard-lil/perma#2198]
Maintenance mode is triggered by nginix observing that a file called maintenance_on.html is present in Perma's static directory. When it sees that "flag", it proceeds to serve maintenance.html (not maintenance_on.html) for all requests, for users not on the VPN.

To do:

- [x] Change the fab task so that it touches and deletes maintenance_on (note lack of file extension), instead of cp maintenance.html maintenance_on.html, which is misleading.

- [x] Add a new feature: we want to be able to serve a custom error page, if it's present, and fall back to maintenance.html if it's not, so that we can use this mechanism both for routine maintenance and if we need to inform users that we are down.

- [ ] We want to be able to trigger this even if the application server is down entirely. Figure out how to make that possible.

**Comment by rebeccacremona** [comment]
re: "We want to be able to trigger this even if the application server is down entirely. Figure out how to make that possible." 

I did a little experimenting with this. It's not possible with Cloudflare config alone (though other users have also requested an easy-to-trigger "maintenance mode" feature. I tried putting up a page using github pages (https://rebeccacremona.github.io/perma/, https://github.com/rebeccacremona/perma/tree/gh-pages) and using Cloudflare page rules to redirect there. That worked, but meant that every page returned status code 200.... which is a deal breaker. We could consider redirecting to a little heroku app, or a little app at lil.tools whose only job is to serve as "we're down!" pages for lil apps..... 

But. Not worth it at this time.
